### PR TITLE
Quartz - improve the error message when scheduler is disabled and org.quartz.Scheduler is injected

### LINKED ISSUE #10633: Create a better exception message when Quartz is used
**Description**

With a pure programmatic Quartz approach, with a code like:

```
@ApplicationScoped
public class PeriodicTask {

@Inject
Scheduler quartz;

public void start() throws SchedulerException {
final JobDetail job = newJob(MyJob.class)
.withIdentity("questions", "millionaire")
.build();

final SimpleTrigger trigger = newTrigger()
.withIdentity("questionsTrigger", "millionaire")
.startNow()
.withSchedule(
SimpleScheduleBuilder.simpleSchedule()
.withIntervalInSeconds(20)
.withRepeatCount(5))
.build();

quartz.getListenerManager()
.addTriggerListener(new EndTriggerListener("EndTrigger"),
keyEquals(triggerKey("questionsTrigger", "millionaire")));
quartz.scheduleJob(job, trigger);
}
}
```
You get the next exception:

```
Caused by: java.lang.IllegalStateException: Cannot produce org.quartz.Scheduler - Quartz scheduler is disabled or no schedules were found
	at io.quarkus.quartz.runtime.QuartzScheduler.produceQuartzScheduler(QuartzScheduler.java:69)
```

When you add the next property:

```
quarkus.quartz.force-start=true
```

Then everything works as expected, so I think that it would be nice to have a better error message saying, I couldn't generate the Scheduler, maybe you have not set the property to true? So people can figure out quickly what is the problem.

### PR DESCRIPTION
**Pull request by mkouba**
... and org.quartz.Scheduler is injected
- resolve #10633

The error message is: `java.lang.IllegalStateException: Quartz scheduler is either explicitly disabled through quarkus.scheduler.enabled=false or no @Scheduled methods were found. If you only need to schedule a job programmatically you can force the start of the scheduler via quarkus.quartz.force-start=true`.

### CODE REVIEW

**Review by lordofthejars — APPROVED** [review]

**Review by geoand — APPROVED** [review]

**Review by machi1990 — APPROVED** [review]

File: docs/src/main/asciidoc/scheduler-reference.adoc
Comment (gsmet) [inline docs/src/main/asciidoc/scheduler-reference.adoc:210]: I'm not entirely sure how this link will be rendered in the HTML. We need it to end up being `quartz` without any extension.

Could you check?

**Review by gsmet — DISMISSED** [review]
Looks good to me. I just added a small comment about the doc. Could you check and dismiss the review one sorted out?

Thanks!

File: docs/src/main/asciidoc/scheduler-reference.adoc
Comment (mkouba) [reply docs/src/main/asciidoc/scheduler-reference.adoc:210]: It renders just fine. Actually, it's the only way I found to link a section from a different document. See https://asciidoctor.org/docs/user-manual/#inter-document-cross-references.

Comment (gsmet) [reply docs/src/main/asciidoc/scheduler-reference.adoc:210]: My fear was that it renders with `.html` at the end which would not help with SEO.

Comment (gsmet) [reply docs/src/main/asciidoc/scheduler-reference.adoc:210]: Yeah and that's exactly what happens unfortunately.

Comment (gsmet) [reply docs/src/main/asciidoc/scheduler-reference.adoc:210]: I think I fixed the issue here: https://github.com/quarkusio/quarkusio.github.io/pull/661 .

But if another pair of eyes can check that I didn't break something else, that would be welcome!

**Pull request merged**
